twenty-first/src/util_types/merkle_tree.rs

```diff
@@ -31,30 +34,36 @@ where
     _hasher: PhantomData<H>,
 }
 
-/// # Design
-/// Static methods are called from the verifier, who does not have
-/// the original `MerkleTree` object, but only partial information from it,
-/// in the form of the quadruples: `(root_hash, index, digest, auth_path)`.
-/// These are exactly the arguments for the `verify_*` family of static methods.
 impl<H> MerkleTree<H>
 where
     H: AlgebraicHasher,
 {
-    /// Given a list of leaf indices, return the indices of exactly those nodes that are needed to
-    /// prove (or verify) that the indicated leaves are in the Merkle tree.
-    // This function is not defined as a method (taking self as argument) since it's
-    // needed by the verifier who does not have access to the Merkle tree.
+    const MAX_NUM_LEAVES: usize = usize::MAX / 2;
```

Comment:
I guess in the Triton VM context, the max number of leaves is $2^{31}$. And it should probably always be a power of two which is not enforced here, no?

Reply:
Good points. :+1: I think setting it to $2^{31}$ makes sense. Depending on the STARK's security parameters, the actual number of nodes usable in Triton VM might be fewer; however, in my opinion, twenty-first should not concern itself with that level of detail.